src/errors.rs

```diff
@@ -110,3 +122,8 @@ impl From<TimestampError> for Error {
         Error::TimestampConv(error)
     }
 }
+
+const _: () = {
+    const fn assert_send_sync<T: Send + Sync>() {}
+    assert_send_sync::<Error>();
+};
```

Comment:
What is this?

Reply:
It asserts that the `Error` type implements `Send + Sync` at compile time.

Reply:
Can we do this in a test rather than in the source code?  And leave documentation to that effect?

Reply:
This is just FYI, I don't particularly oppose moving it to a test if that's what you want to do.

Since these are auto traits, it could mean that the `Error` type loses its `Send + Sync`-ness due to an upstream change. You might want to keep this as an earlier, more appropriately placed compilation error to catch that.

Reply:
Well, let's at least document what it is doing